pkg/dao/snapshots_test.go

```diff
@@ -1030,3 +1031,44 @@ func (s *SnapshotsSuite) TestBulkDeleteNotFound() {
 	assert.True(t, slices.ContainsFunc(snaps, func(snap models.Snapshot) bool { return snap.UUID == s1.UUID }))
 	assert.True(t, slices.ContainsFunc(snaps, func(snap models.Snapshot) bool { return snap.UUID == s3.UUID }))
 }
+
+func (s *SnapshotsSuite) TestSetDetectedOSVersion() {
+	t := s.T()
+	tx := s.tx
+	ctx := context.Background()
+
+	mTangy, origTangy := mockTangy(s.T())
+	defer func() { config.Tang = origTangy }()
+
+	repoConfig := s.createRedhatRepository()
+	snap := s.createSnapshot(repoConfig)
+	assert.Empty(t, snap.DetectedOSVersion)
+
+	expectedPackages := []tangy.RpmListItem{
+		{
+			Name:    "redhat-release",
+			Arch:    "x86_64",
+			Version: "9.4",
+			Release: "0.4.el9",
+			Epoch:   "0",
+			Summary: "Red Hat Enterprise Linux release file",
+		},
```

Comment:
small nitpick on the test, could you add a 2nd rpm for 9.3 (or something similar), just so thre's more than 1 result.